packages/react-vapor/src/components/refresh/RefreshCallback.tsx

```diff
@@ -0,0 +1,105 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {Defaults} from '../../Defaults';
+import {IReactVaporState} from '../../ReactVapor';
+import {IDispatch} from '../../utils/ReduxUtils';
+import {RefreshCallBackActions} from './RefeshCallbackActions';
+import {RefreshStatus, RefreshStatusSelectors} from './RefreshCallbackReducer';
+
+export interface IRefreshCallbackOwnProps {
+    id: string;
+    delay?: number;
+    renderCount?: (count: number) => React.ReactNode;
+    callback: (start: () => void) => void;
+}
+
+export interface IRefreshCallbackState {
+    count: number;
+}
+
+const mapStateToProps = (state: IReactVaporState, ownProps: IRefreshCallbackOwnProps) => ({
+    status: RefreshStatusSelectors.getRefreshStatus(state, {id: ownProps.id}),
+});
+
+const mapDispatchToProps = (dispatch: IDispatch, ownProps: IRefreshCallbackOwnProps) => ({
+    start: () => dispatch(RefreshCallBackActions.start(ownProps.id)),
+    stop: () => dispatch(RefreshCallBackActions.stop(ownProps.id)),
+    inProgress: () => dispatch(RefreshCallBackActions.inProgress(ownProps.id)),
+});
+
+export type IRefreshCallbackProps = IRefreshCallbackOwnProps &
+    ReturnType<typeof mapStateToProps> &
+    ReturnType<typeof mapDispatchToProps>;
+
+class RefreshCallbackDisconnected extends React.PureComponent<IRefreshCallbackProps, IRefreshCallbackState> {
+    static defaultProps: Partial<IRefreshCallbackProps> = {
+        delay: 10,
+        renderCount: (count: number) => <span>Auto refresh in {count} seconds</span>,
+    };
+
+    private activeInterval: number;
+
+    constructor(props: IRefreshCallbackProps) {
+        super(props);
+        this.state = {
+            count: this.props.delay,
+        };
+    }
```

Comment:
You might be able to replace the constructor with:
```typescript
state = {
  count: this.props.delay,
};
```

Reply:
If I add the state as param for the constructor. I think I can do that. I try